src/x/errors/errors.go

```diff
@@ -130,6 +131,26 @@ func GetInnerInvalidParamsError(err error) error {
 	return nil
 }
 
+// IsTimeoutError checks if the error is or contains a context.Canceled / context.DeadlineExceeded
+func IsTimeoutError(err error) bool {
```

Comment:
might be nice to make this more generic and the same signature as the std lib `func Is(err, target error) bool`

then you could have a wrapper 
```go
func IsTimeoutError(err error) bool {
  return Is(err, Canceled) || Is(DeadlineExceeded)
}
```